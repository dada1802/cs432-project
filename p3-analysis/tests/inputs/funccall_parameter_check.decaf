def int add2(int a)
{
    return a + 2;
}

def int subtract(int a, int b)
{
    return a - b;
}

def int main()
{
    int a;
    int b;
    int c;
    a = add2(false);
    b = add2(2, 4);
    c = subtract(a);

    return 0;
}