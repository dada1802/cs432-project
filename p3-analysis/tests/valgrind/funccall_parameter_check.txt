==3079902== Memcheck, a memory error detector
==3079902== Copyright (C) 2002-2017, and GNU GPL'd, by Julian Seward et al.
==3079902== Using Valgrind-3.18.1 and LibVEX; rerun with -h for copyright info
==3079902== Command: ../decaf inputs/funccall_parameter_check.decaf
==3079902== 
Type mismatch in parameter 0 of call to 'add2': expected int but found bool on line 16
Invalid number of function arguments on line 17
Invalid number of function arguments on line 18
==3079902== 
==3079902== HEAP SUMMARY:
==3079902==     in use at exit: 0 bytes in 0 blocks
==3079902==   total heap usage: 2,116 allocs, 2,116 frees, 260,088 bytes allocated
==3079902== 
==3079902== All heap blocks were freed -- no leaks are possible
==3079902== 
==3079902== For lists of detected and suppressed errors, rerun with: -s
==3079902== ERROR SUMMARY: 0 errors from 0 contexts (suppressed: 0 from 0)
