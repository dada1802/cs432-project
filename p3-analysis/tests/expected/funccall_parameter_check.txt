Type mismatch in parameter 0 of call to 'add2': expected int but found bool on line 16
Invalid number of function arguments on line 17
Invalid number of function arguments on line 18
